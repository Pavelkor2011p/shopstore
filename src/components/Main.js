import React, { Component } from 'react'

export class Main extends Component {
    render (){
    return (
    <main className="main conteiner">
        <div className="main__content">
            <a href="/" className="main__content-home">Главная</a>
            <p className="main__content-cosm">Косметика и гигиена</p>
        </div>
        <div className="main__top">
            <h1 className="main__top-title">Косметика и гигиена</h1>
            <p className="main__top-select">Сортировка:  <select>
            <option>Название</option>
            <optgroup label="Цена">
                <option value="">По возрастанию</option>
                <option value="">По убыванию</option>
            </optgroup>
            </select></p>
        </div>
        </main>
    )
    }
}
export default Main