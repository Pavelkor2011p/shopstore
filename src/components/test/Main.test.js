import{render, screen} from '@testing-library/react';
import Main from '../Main';

describe('Main render', () => {
    it('should create Main with empty main', () => {
        const utils = render(<Main />);
        expect(utils).toMatchSnapshot();
    })
    it('list renders', ()=> {
        render(<Main />);
        const Element = screen.getAllByText(/Главная/i)
        expect(Element).toMatchSnapshot();
    
    })

    it('title renders', ()=> {
        render(<Main />);
        const Element = screen.getAllByText(/Косметика/i)
        expect(Element).toMatchSnapshot();
        
    })

    it('option renders', ()=> {
        render(<Main />);
        const Element = screen.getAllByText(/убыванию/i)
        expect(Element).toMatchSnapshot();
        
    })
})