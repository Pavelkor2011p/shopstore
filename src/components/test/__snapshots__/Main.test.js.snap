// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render list renders 1`] = `
Array [
  <a
    class="main__content-home"
    href="/"
  >
    Главная
  </a>,
]
`;

exports[`Main render option renders 1`] = `
Array [
  <option
    value=""
  >
    По убыванию
  </option>,
]
`;

exports[`Main render should create Main with empty main 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main conteiner"
      >
        <div
          class="main__content"
        >
          <a
            class="main__content-home"
            href="/"
          >
            Главная
          </a>
          <p
            class="main__content-cosm"
          >
            Косметика и гигиена
          </p>
        </div>
        <div
          class="main__top"
        >
          <h1
            class="main__top-title"
          >
            Косметика и гигиена
          </h1>
          <p
            class="main__top-select"
          >
            Сортировка:  
            <select>
              <option>
                Название
              </option>
              <optgroup
                label="Цена"
              >
                <option
                  value=""
                >
                  По возрастанию
                </option>
                <option
                  value=""
                >
                  По убыванию
                </option>
              </optgroup>
            </select>
          </p>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main conteiner"
    >
      <div
        class="main__content"
      >
        <a
          class="main__content-home"
          href="/"
        >
          Главная
        </a>
        <p
          class="main__content-cosm"
        >
          Косметика и гигиена
        </p>
      </div>
      <div
        class="main__top"
      >
        <h1
          class="main__top-title"
        >
          Косметика и гигиена
        </h1>
        <p
          class="main__top-select"
        >
          Сортировка:  
          <select>
            <option>
              Название
            </option>
            <optgroup
              label="Цена"
            >
              <option
                value=""
              >
                По возрастанию
              </option>
              <option
                value=""
              >
                По убыванию
              </option>
            </optgroup>
          </select>
        </p>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Main render title renders 1`] = `
Array [
  <p
    class="main__content-cosm"
  >
    Косметика и гигиена
  </p>,
  <h1
    class="main__top-title"
  >
    Косметика и гигиена
  </h1>,
]
`;
